### PR TITLE
cdc/scheduler_v2: interfaces and mocks for ScheduleDispatcher

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
- In preparation to merge the ScheduleDispatcher, we need to clearly define the interfaces involved.

### What is changed and how it works?
- Defined interface `ScheduleDispatcher`.
- Defined interface `ScheduleDispatcherCommunicator`.
- Implemented `mockScheduleDispatcherCommunicator` for testing.

### Check List 

Tests 

 - No code (no unit-testable code)

Code changes

 - Has exported function/method change

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/scheduler/scheduler_dispatcher.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	DispatchTable(ctx context.Context,
+		changeFeedID model.ChangeFeedID,
+		tableID model.TableID,
+		captureID model.CaptureID,
+		isDelete bool, // True when we want to remove a table from the capture.
+	) (done bool, err error)
+
+	// Announce announces to the specified capture that the current node has become the Owner.
+	Announce(ctx context.Context,
+		changeFeedID model.ChangeFeedID,
```
Comment (amyangfei) [inline cdc/scheduler/scheduler_dispatcher.go:56]: Why does this API need to pass a changefeed ID

Comment (liuzix) [reply cdc/scheduler/scheduler_dispatcher.go:56]: A changefeed-independent `Announce` seems a little bit hard to implement without changing too many components of our current Owner implementation. Changefeed-wise `Announce` makes the protocol easier to reason about and maintains the structure of the original per-changefeed scheduler.

In our short- to mid- term road map, we do not seem to have an urgent requirement for 10000+ changefeeds, so the performance overhead seems acceptable.

File: cdc/scheduler/scheduler_dispatcher_test.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import (
+	"github.com/pingcap/log"
+	"github.com/pingcap/ticdc/cdc/model"
+	cdcContext "github.com/pingcap/ticdc/pkg/context"
+	"github.com/stretchr/testify/mock"
+	"go.uber.org/zap"
+)
+
+type mockScheduleDispatcherCommunicator struct {
```
Comment (ben1009) [inline cdc/scheduler/scheduler_dispatcher_test.go:24]: seems the mock is modified after generated, any rules should fellow if someone else modify later

File: cdc/scheduler/scheduler_dispatcher.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+		captureID model.CaptureID) (done bool, err error)
+}
+
+const (
```
Comment (3AceShowHand) [inline cdc/scheduler/scheduler_dispatcher.go:60]: can you move this const to the top of the file?

File: cdc/scheduler/scheduler_dispatcher_test.go (hunk 2 above)
Comment (liuzix) [reply cdc/scheduler/scheduler_dispatcher_test.go:26]: This mock is not generated. Do you have any specific suggestions about the rules?

**Review by amyangfei — APPROVED** [review]

**Review by 3AceShowHand — APPROVED** [review]

**Pull request merged**